### PR TITLE
Catch em all

### PR DESCRIPTION
**Pull request by privat**
Extends the `do catch` construct to catch all run time errors.
The interpreter also keep the error information if one wants it.

### CODE REVIEW

File: src/compiler/abstract_compiler.nit [hunk 1]
```diff
@@ -1825,17 +1825,30 @@ abstract class AbstractCompilerVisitor
...
 	fun add_abort(message: String)
 	do
+		add_raw_throw
+		self.add("PRINT_ERROR(\"Runtime error: %s\", \"{message.escape_to_c}\");")
+		add_raw_abort
+	end
+
+	# Generate a long jump if there is a catch block.
+	#
+	# This method shlould be called before the error messages before a `add_raw_abort`.
```
Comment (jcbrinfo) [inline src/compiler/abstract_compiler.nit]: > This method shlould be called before the error messages **and** before a `add_raw_abort`

File: src/compiler/abstract_compiler.nit [hunk 2]
```diff
@@ -1825,17 +1825,30 @@ abstract class AbstractCompilerVisitor
...
 		self.add("longjmp(catchStack.envs[catchStack.cursor], 1);")
 		self.add("\}")
-		self.add("PRINT_ERROR(\"Runtime error: %s\", \"{message.escape_to_c}\");")
-		add_raw_abort
 	end
 
+	# Generate abort without a message.
+	#
+	# Used when one need a more complex message.
+	# Do not forget to call add_raw_abort before the custom user message.
```
Comment (jcbrinfo) [inline src/compiler/abstract_compiler.nit]: > Do not forget to call **\`add_raw_throw\`** before the **display of a** custom user message.

**Comment by privat** [comment]
reroll: spell

File: src/interpreter/naive_interpreter.nit [hunk 3]
```diff
@@ -119,7 +119,10 @@ class NaiveInterpreter
 	var escapemark: nullable EscapeMark = null
 
 	# The count of `catch` blocs that have been encountered and can catch an abort
-	var catch_count = 0
+	var catch_count = 0 is writable
```
Comment (ppepos) [inline src/interpreter/naive_interpreter.nit]: Is this still relevant? I see nowhere where it is set in this PR.

**Pull request merged**